### PR TITLE
fix(aws-cdk-lib): ADOT Lambda Layer constants update - v0.32.0

### PR DESCRIPTION
**Pull request by vasireddy99**
Update constants/doc for ADOT Lambda Layer
Reference PR - #26793 

**NOTE**  This PR requires an update to an integration test and am unable
to run the `cdk-integ` tool to perform a real deployment

----

*By submitting this pull request, I confirm that I make my contribution under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-lambda/lib/adot-layers.ts [hunk 1]
```diff
@@ -220,6 +220,11 @@ export class AdotLambdaLayerJavaSdkVersion extends AdotLambdaLayerVersion {
    */
   public static readonly V1_19_0 = new AdotLambdaLayerJavaSdkVersion('1.19.0');
 
+  /**
+   * Version 1.28.1
+   */
+  public static readonly V1_28_1 = new AdotLambdaLayerJavaSdkVersion('1.28.1');
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-lambda/lib/adot-layers.ts:226]: It's better to follow the convention of `AdotLambdaLayerPythonSdkVersion` and list the versions in descending order.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks for the contribution! 
Shouldn't the `LATEST` versions be updated as well?
If so, [this](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk-testing/framework-integ/test/aws-lambda/test/integ.lambda-adot.ts) integration test snapshot will need to be updated accordingly.

**Review by lpizzinidev — COMMENTED** [review]
[This](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk-testing/framework-integ/test/aws-lambda/test/integ.lambda-adot.ts) integration test template needs to be updated to match the new `LATEST` versions.

**Comment by vasireddy99** [comment]
I am seeking for Clarification Request, Also added a note in the PR description

**Review by lpizzinidev — APPROVED** [review]

**Review by mrgrain — APPROVED** [review]

**Pull request merged**